core/model/modx/processors/resource/delete.class.php

```diff
@@ -256,12 +256,16 @@ public function fireAfterDelete(array $childrenIds = array()) {
      * @return void
      */
     public function clearCache() {
-        $this->modx->cacheManager->refresh(array(
-            'db' => array(),
-            'auto_publish' => array('contexts' => array($this->resource->get('context_key'))),
-            'context_settings' => array('contexts' => array($this->resource->get('context_key'))),
-            'resource' => array('contexts' => array($this->resource->get('context_key'))),
-        ));
+		$clear = $this->getProperty('syncsite',$this->modx->getOption('syncsite_default')) || $this->getProperty('clearCache',false);
+        if ($clear) {
+            $this->modx->cacheManager->refresh(array(
+                'db' => array(),
+                'auto_publish' => array('contexts' => array($this->workingContext->get('key'))),
+                'context_settings' => array('contexts' => array($this->workingContext->get('key'))),
+                'resource' => array('contexts' => array($this->workingContext->get('key'))),
+            ));
+        }
+        return $clear;
```

Comment:
According to the docblock comment for this function it should return void. Either the docblock needs to be adjusted or this needs to be changed to return void.